community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -131,6 +132,7 @@ public final void close()
             if ( cursor.pagedFile != null )
             {
                 cursor.unpinCurrentPage();
+                cursor.pageCursorTracer.reportEvents();
```

Comment:
There are couple of reasons why i did it like that:
- if statement is quite big and long running we will not gonna be able to see any page cache counters update and that will make tracing and understanding current situation non clear and hard to trace since all counters will be reported only once in the end when statement will be closed.
- currently code is really bad on closing statement (that is a some code path problem and should be fixed)

Agree on the point of linked cursors: if we will assume that our page cursors tracers are thread local and all of the cursors have the same tracer we should avoid doing reporting multiple times.
My thinking was a bit different - to add possibility to store your local counters in cursor tracers for statement tracing later on.

But agree your proposal will make code a bit simpler and faster even if two mentioned above points will not gonna be solved, but we can come back to them later if that will be a problem. Will do like you suggest.